src/main/java/org/terasology/kcomputers/rendering/nui/layers/ComputerDisplayWidget.java

```diff
@@ -69,14 +69,37 @@ public boolean onKeyEvent(NUIKeyEvent event) {
         EntityRef ref = displayComponent.get().getEntityRef();
         if (ref.hasComponent(KallistiKeyboardComponent.class) && TranslationAWTLWJGL.hasLwjgl(event.getKey().getId())) {
             localPlayer.get().send(new KallistiKeyPressedEvent(
-                new KeyboardInputProvider.Key(
-                    event.isDown() ? KeyboardInputProvider.KeyType.PRESSED : KeyboardInputProvider.KeyType.RELEASED,
-                    TranslationAWTLWJGL.toAwt(event.getKey().getId()),
-                    event.isDown() ? event.getKeyCharacter() : lastCharacter
-                )
+                    new KeyboardInputProvider.Key(
+                            event.isDown() ? KeyboardInputProvider.KeyType.PRESSED :
+                                    KeyboardInputProvider.KeyType.RELEASED,
+                            TranslationAWTLWJGL.toAwt(event.getKey().getId()),
+                            event.isDown() ? event.getKey().getId() : lastCharacter
+                    )
             ));
             if (event.isDown()) {
-                lastCharacter = event.getKeyCharacter();
+                lastCharacter = event.getKey().getId();
+            }
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean onCharEvent(NUICharEvent event) {
+
+        EntityRef ref = displayComponent.get().getEntityRef();
+        char character = event.getCharacter();
+        if (ref.hasComponent(KallistiKeyboardComponent.class) && TranslationAWTLWJGL.hasLwjgl(character)) {
+            localPlayer.get().send(new KallistiKeyPressedEvent(
+                    new KeyboardInputProvider.Key(
+                            event.getKeyboard().isKeyDown(character) ? KeyboardInputProvider.KeyType.PRESSED :
+                                    KeyboardInputProvider.KeyType.RELEASED,
+                            TranslationAWTLWJGL.toAwt(character),
```

Comment:
Try to hardcode with awt's VK key (suggest VK_STOP, for fun, or another unused key)

Reply:
Why should I hardcode a key here instead of using the event's character?

Reply:
You haven't character there, only keycode(this is code for keyboard key)
P.s. we used lwjgl2's keycode definition. Lwjgl3 PR only.map glfw's codes to lwjgl2